src/Trace/TracerFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace OpenTelemetry\Trace;
+
+use InvalidArgumentException;
+use OpenTelemetry\Context\SpanContext;
+use OpenTelemetry\Trace\SpanProcessorInterface;
+
+class TracerFactory
+{
+    /**
+     * @var self
+     */
+    protected static $instance;
+
+    /**
+     * @var Tracer[]
+     */
+    protected $tracers;
+
+    /**
+     * @var SpanProcessorInterface[]
+     */
+    protected $spanProcessors;
+
+    private final function __construct()
+    {
```

Comment:
What are you planning on adding in this constructor?  I would assume you are planning on creating an object containing some of the included vars?

Reply:
I didn't plan to do anything, just wanted to implement TracerFactory as a singleton. I took the suggestion from specification: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#obtaining-a-tracer

Now I think that I can use the constructor to inject SpanProcessors and other configuration in the TracerFactory object and not in getInstance() method.

Reply:
Sounds like a good plan.